src/Microsoft.DotNet.Interactive/InputTextStream.cs

```diff
@@ -58,13 +63,15 @@ private void EnsureStarted()
                         _channel.OnNext(line);
                     }
                 }
-            });
+            }, _cancellationSource.Token);
         }
 
         public void Dispose()
         {
+            _channel.OnNext(string.Empty);
```

Comment:
This looks odd. Why is this needed?

Reply:
Completing that steam without any value causes LastAsync to throw as there are now value for the async subject